hashes/src/sha256.rs

```diff
@@ -336,7 +344,7 @@ impl Midstate {
         w
     }
 
-    const fn const_hash(bytes: &[u8], finalize: bool) -> Self {
+    const fn compute_midstate_unoptimized(bytes: &[u8], finalize: bool) -> Self {
```

Comment:
This is not really a great name because it can also output hash. Maybe `perform_hashing_unoptimized`?